### ISSUE TITLE
saving pet api does't work

### ISSUE THREAD

**Issue by cosmosof** [awslabs/aws-mobile-react-native-starter#34]
I am not able to save new pet. 
`    return await API.post('Pets','/items/pets',{headers:{'content-type': 'application/json; charset=utf-8'}});
`
this line not working

error saving pet... Error: Request failed with status code 400

**Comment by filanthropic** [comment]
Having the same issue.

**Comment by filanthropic** [comment]
To fix, replace the existing apiSavePet function with the following in AddPet.js:
```
async apiSavePet(pet) {
    let myInit = {
      body: pet,
      headers: {'content-type': 'application/json; charset=utf-8'}
    }
    
    return await API.post('Pets','/items/pets',myInit);
}
```

**Comment by twoolf** [comment]
This change is consistent with the docs (https://github.com/aws/aws-amplify/blob/master/media/api_guide.md#post), but I'm still not getting it to save -- I am wondering if it is due to this error: https://github.com/aws/aws-amplify/issues/69

The S3 image save works and the API.post error doesn't seem to imply a problem with authentication or policies, so it does seem most likely a problem within the SDK.  Is anyone else having problems with the current build not being able to save to DynamoDB?

I'm thinking it would be best to go back to previous versions: any suggestions on what to dial back first?

**Comment by mlabieniec** [comment]
@twoolf A couple things here:
 
1. Make sure you have updated your API gateway with your custom headers (and CORS) as outlined [here](https://github.com/aws/aws-amplify/blob/master/media/api_guide.md#note-about-headers). 
2. This type of error (signature) usually means that either the URL (/items/etc) is incorrect and not present on the API. Sometimes this occurs when trying to send query parameters on the url like /items?test=1. This would avail the same error since the signature that's calculated will not match API gateway (which is by default a lambda proxy if you use the awsmobile-cli to generate it)

There hasn't been any significant updates/changes to the API category in the last couple sprints so i believe your issue might be related to the header not being present/configured in your API gateway CORS. The best way to debug this is to enable CloudWatch Logs in your API Gateway stage and enable full (info) request logs. That way you will be able to see the request and why it's being rejected within CloudWatch logs.

**Comment by jrounsav** [comment]
@twoolf I'm seeing the same problem when posting with a body. If I remove the body I am able to post to my test endpoints.
Will check back after I finish working through that piece of the signing process. Something in there appears to be borked.

**Comment by manueliglesias** [comment]
@twoolf, @jrounsav The react native starter has included some changes in #39 to address this, in particular to fix the signing process (See aws/aws-amplify#128 and aws/aws-amplify#129)

Please try with the latest versions of both the starter and the aws-amplify-react-native packages and let us know how it goes.

**Comment by twoolf** [comment]
confirming that it now works -- thanks! -- for anyone else having problems with a new build from this codebase, the other trick needed, at least on my end, is from this: https://github.com/facebook/react-native/issues/14209
 : yarn upgrade react-native
since otherwise it was crashing at 97% of build, even following the suggestions about emptying cache and/or restarts; probably this won't be needed for most, since this step is tied more directly to the react-native code --
